Source/WebCore/editing/FrameSelection.h

```diff
@@ -268,6 +272,8 @@ class FrameSelection : private CaretBase {
     void disassociateLiveRange();
     void updateFromAssociatedLiveRange();
 
+    std::unique_ptr<CaretAnimation>& currentCaretAnimation() { return m_currentCaretAnimation; }
```

Comment:
Is it ever non-null? Can you return a reference?

Reply:
This should return CaretAnimation* or CaretAnimation&, not unique_ptr.